apps/server/src/modules/board/uc/submission-item.uc.ts

```diff
@@ -12,41 +12,76 @@ export class SubmissionItemUc {
 		private readonly elementService: ContentElementService,
+		private readonly submissionItemService: SubmissionItemService,
 		private readonly logger: Logger
 	) {
 		this.logger.setContext(SubmissionItemUc.name);
 	}
 
 	async findSubmissionItems(userId: EntityId, submissionContainerId: EntityId): Promise<SubmissionItem[]> {
-		const submissionContainer = (await this.elementService.findById(
-			submissionContainerId
-		)) as SubmissionContainerElement;
-
-		if (!(submissionContainer instanceof SubmissionContainerElement))
-			throw new HttpException('Id does not belong to a submission container', HttpStatus.UNPROCESSABLE_ENTITY);
+		const submissionContainer = await this.getSubmissionContainer(submissionContainerId);
+		await this.checkPermission(userId, submissionContainer, Action.read);
 
-		const submissionItems = submissionContainer.children as SubmissionItem[];
+		let submissionItems = submissionContainer.children as SubmissionItem[];
 
-		if (!submissionItems.every((child) => child instanceof SubmissionItem))
+		if (!submissionItems.every((child) => child instanceof SubmissionItem)) {
 			throw new HttpException(
 				'Children of submission-container-element must be of type submission-item',
 				HttpStatus.UNPROCESSABLE_ENTITY
 			);
+		}
 
-		await this.checkPermission(userId, submissionContainer, Action.read);
-
-		if (await this.isStudent(userId, submissionContainer)) {
-			return submissionItems.filter((item) => item.userId === userId);
+		const isAuthorizedStudent = await this.isAuthorizedStudent(userId, submissionContainer);
+		if (isAuthorizedStudent) {
+			submissionItems = submissionItems.filter((item) => item.userId === userId);
 		}
 
 		return submissionItems;
 	}
 
-	private async isStudent(userId: EntityId, boardDo: AnyBoardDo): Promise<boolean> {
+	async updateSubmissionItem(
+		userId: EntityId,
+		submissionItemId: EntityId,
+		completed: boolean
+	): Promise<SubmissionItem> {
+		const submissionItem = await this.submissionItemService.findById(submissionItemId);
+
+		await this.checkPermission(userId, submissionItem, Action.read, UserRoleEnum.STUDENT);
+		if (submissionItem.userId !== userId) {
+			throw new ForbiddenException();
+		}
+
+		await this.submissionItemService.update(submissionItem, completed);
+
+		return submissionItem;
+	}
+
+	private async isAuthorizedStudent(userId: EntityId, boardDo: AnyBoardDo): Promise<boolean> {
 		const boardDoAuthorizable = await this.boardDoAuthorizableService.getBoardAuthorizable(boardDo);
 		const userRoleEnum = boardDoAuthorizable.users.find((u) => u.userId === userId)?.userRoleEnum;
-		if (!userRoleEnum) throw new ForbiddenException('User not part of this board');
-		return userRoleEnum === UserRoleEnum.STUDENT;
+
+		if (!userRoleEnum) {
+			throw new ForbiddenException('User not part of this board');
+		}
+
+		// TODO do this with permission instead of role and using authorizable rules
+		if (userRoleEnum === UserRoleEnum.STUDENT) {
+			return true;
+		}
+
+		return false;
```

Comment:
to avoid using return in if-statement you could use:
		// if (userRoleEnum === UserRoleEnum.STUDENT) {
		// 	return true;
		// }
		// return false;

		const isAuthorizedStudent = userRoleEnum === UserRoleEnum.STUDENT;

		return isAuthorizedStudent;